### PR TITLE
Amazon integration

### LINKED ISSUE #3: Amazon API
Research features and limitations of Amazon API and create basic classes for getting important data

### PR DESCRIPTION
**Pull request by guser21**
Resolve #3

### CODE REVIEW

File: server/build.gradle [hunk 1]
```diff
@@ -32,4 +32,7 @@ bootJar {
 dependencies {
 	implementation('org.springframework.boot:spring-boot-starter-web')
 	testImplementation('org.springframework.boot:spring-boot-starter-test')
+	// https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-iam
```
Comment (Ogochi) [inline server/build.gradle]: ?

File: server/src/main/java/com/omigost/server/iam/impl/AIMProvider.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagement;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagementClientBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class AIMProvider {
+    @Value("aws-region")
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/iam/impl/AIMProvider.java]: Shouldn't it be "${aws-region}" ?

File: server/src/main/java/com/omigost/server/iam/impl/IAMServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+public class IAMServiceImpl implements IAMService {
+    private AmazonIdentityManagement iam;
+    @Override
+    public List<AwsUser> users() {
+
+        boolean gotAllUsers = false;
+        List<AwsUser> resultList = new ArrayList<>();
+        ListUsersRequest request = new ListUsersRequest();
+
+        //results are paginated
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/iam/impl/IAMServiceImpl.java]: space after "//"

File: server/src/main/java/com/omigost/server/model/AwsUser.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+//better nameing
+@Data
+@Builder
+public class AwsUser {
+    AwsUser manager;
+    String awsPath;
+    String awsUserName;
+    String awsUserId;
+    String arn; //amazon resource name
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/model/AwsUser.java]: space after "//"

File: server/src/main/java/com/omigost/server/iam/impl/IAMServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        while (!gotAllUsers) {
+            ListUsersResult response = iam.listUsers(request);
+            List<User> users = response.getUsers();
+
+            List<AwsUser> awsUsers = users.stream()
+                    .map(user -> AwsUser.builder()
+                            .arn(user.getArn())
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/iam/impl/IAMServiceImpl.java]: Please style it in better way

File: server/src/main/java/com/omigost/server/model/AwsUser.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.omigost.server.model;
+
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.List;
+
+//better nameing
```
Comment (Ogochi) [inline server/src/main/java/com/omigost/server/model/AwsUser.java]: space after "//" and typo

**Comment by Ogochi** [comment]
Additionally change PR comment to make it resolve issue: "Resolves #ISSUE_NUMBER".

File: server/build.gradle (hunk 1 above)
Comment (mbalc) [reply server/build.gradle]: please elaborate

File: server/src/main/java/com/omigost/server/iam/impl/AIMProvider.java (hunk 2 above)
Comment (mbalc) [reply server/src/main/java/com/omigost/server/iam/impl/AIMProvider.java]: Yeah, we should probably use ENV for that

**Comment by mbalc** [comment]
> Additionally change PR comment to make it resolve issue: `Resolves #ISSUE_NUMBER`.
> Also fix commits - use `git rebase` instead of merging.

@Ogochi Actually it looks like you rewrote the whole master branch for some reason - we should avoid doing that at all costs in the future because, like, there's no reason to do that

**Comment by Ogochi** [comment]
@mbalc That's true I rewrote master, but I have done it just after mini hackathon to cleanup fix commits. I understand concept and I'm not going to do anything like that in the future.

**Pull request merged**